consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -70,19 +70,65 @@ where
         Self { branches, tips }
     }
 
+    /// Create a new [`Branches`] instance with the updated state.
+    /// This method adds a [`Branch`] to the new instance without running
+    /// validation.
+    ///
+    /// # Warning
+    ///
+    /// **This method bypasses safety checks** and can corrupt the state if used
+    /// incorrectly.
+    /// Only use for recovery, debugging, or other manipulations where the input
+    /// is known to be valid.
+    ///
+    /// # Arguments
+    ///
+    /// * `header` - The ID of the block to be added.
+    /// * `parent` - The ID of the parent block. Due to the nature of the method
+    ///   (`unchecked`), the existence of the parent block is not verified.
+    /// * `slot` - The slot of the block to be added.
+    /// * `length`: The position of the block in the chain.
+    #[must_use = "Returns a new instance with the updated state, without modifying the original."]
+    fn apply_header_unchecked(&self, header: Id, parent: Id, slot: Slot, length: u64) -> Self {
+        let mut branches = self.branches.clone();
+
+        // Add the new header to the branches. Parent doesn't need to be removed since
+        // it should not be in the tips set.
+        // Maybe we can run the `tips.remove(parent)` nonetheless (since it would be a
+        // no-op). That would allow us to reuse this method in the
+        // [`Self::apply_header`].
```

Comment:
This seems reasonable. It is basically a duplicated of the below for a specific case in where the below one applies anyway? `apply_header` is already valid for this case.

Reply:
I thought so, but I'm left wondering whether, for the unchecked case, it'd make sense to return some sort of error if, when removing the parent, it does exist in `tips`:
- On the one hand, I think it doesn't make sense, as unchecked would cover that. 
- On the other, the specific use case for unchecked wouldn't have the parent, so its existence could point to something being off.